src/main/java/com/mraof/minestuck/block/redstone/StatStorerBlock.java

```diff
@@ -0,0 +1,105 @@
+package com.mraof.minestuck.block.redstone;
+
+import com.mraof.minestuck.client.gui.MSScreenFactories;
+import com.mraof.minestuck.effects.CreativeShockEffect;
+import com.mraof.minestuck.tileentity.redstone.StatStorerTileEntity;
+import com.mraof.minestuck.util.ParticlesAroundSolidBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.particles.RedstoneParticleData;
+import net.minecraft.state.IntegerProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Direction;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+/**
+ * Outputs a redstone signal proportional to the recorded value its tile entity is currently set to, divided by the divide by value stored also stored in the tile entity.
+ * Stats are increased when the relevant event(in ServerEventHandler) occurs within 16 blocks of a stat storer. GUI is limited by creative shock
+ */
+public class StatStorerBlock extends Block
+{
+	public static final IntegerProperty POWER = BlockStateProperties.POWER;
+	
+	public StatStorerBlock(Properties properties)
+	{
+		super(properties);
+		this.registerDefaultState(stateDefinition.any().setValue(POWER, 0));
+	}
+	
+	@Override
+	@SuppressWarnings("deprecation")
+	public ActionResultType use(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand hand, BlockRayTraceResult hit)
+	{
+		if(worldIn.isClientSide && !player.isCrouching() && !CreativeShockEffect.doesCreativeShockLimit(player, 1))
+		{
+			TileEntity tileEntity = worldIn.getBlockEntity(pos);
+			if(tileEntity instanceof StatStorerTileEntity)
+			{
+				StatStorerTileEntity te = (StatStorerTileEntity) tileEntity;
+				
+				MSScreenFactories.displayStatStorerScreen(te);
+			}
+		}
+		
+		return ActionResultType.SUCCESS;
```

Comment:
It doesn't seem right that you always return success whether you actually interact with the block or not. Look at how vanilla handles the result type, and then either give motivation to why this looks like it does, or change it to something more appropriate.

Reply:
This should be looked at if you have the time. Though it's fine to leave it as a todo in affected places if these block interactions have been decently tested beforehand.